content/docs/user-guide/gems/reference/aws/aws-gamelift/scripting.md

```diff
@@ -0,0 +1,121 @@
+
+After successfully creating a session, other instances of your game can discover it using the **SearchSessions** node and then join it using the **JoinSession** node.
+
+
+#### AWSGameLiftCreateSessionRequest Get and Set variable nodes
+
+![AWSGameLiftCreateSessionRequest Get and Set Variable nodes](/images/user-guide/gems/reference/aws/aws-gamelift/createsessionrequest.PNG)
+
+
+#### AWSGameLiftCreateSessionOnQueueRequest Get and Set variable nodes
+
+![AWSGameLiftCreateSessionOnQueueRequest Get and Set nodes](/images/user-guide/gems/reference/aws/aws-gamelift/createsessiononqueuerequest.PNG)  
+
+
+#### CreateSessionAsync node and Async Handler node
+
+![CreateSessionAsync node and Async Handler node](/images/user-guide/gems/reference/aws/aws-gamelift/createsessionasync.PNG)  
+
+
+#### CreateSession node
+
+![CreateSession node](/images/user-guide/gems/reference/aws/aws-gamelift/createsession.PNG)  
+
+
+### SearchSessions
+
+Get a list of game sessions that are currently active and joinable using the **SearchSessions** node. You can specify your search criteria using the **AWSGameLiftSearchSessionsRequest** node.
+
+On success, the SearchSessions node returns a list of SearchSessionsResponse objects, which provide details of the sessions found that meet your search criteria.
+
+  
+#### AWSGameLiftSearchSessionsRequest Get and Set variable nodes
+
+![AWSGameLiftSearchSessionsRequest Get and Set variable nodes](/images/user-guide/gems/reference/aws/aws-gamelift/searchsessionrequest.PNG)  
+
+
+#### SearchSessionsResponse Get and Set variable nodes
+
+![SearchSessionsResponse Get and Set variable nodes](/images/user-guide/gems/reference/aws/aws-gamelift/searchsessionresponse.PNG)  
+  
+
+#### SearchSessions node
+
+![SearchSessions node](/images/user-guide/gems/reference/aws/aws-gamelift/searchsessions.PNG)  
+  
+
+#### SearchSessionsAsync node and Async Handler node
+
+![SearchSessionsAsync node and Async Handler node](/images/user-guide/gems/reference/aws/aws-gamelift/searchsessionasync.PNG)  
+
+
+### JoinSession
+
+After searching for a session, you can join a session using the `JoinSession` node. Specify the session using its `game session id`, which you retrieved from SearchSession. If the game successfully connects to the server, you will automatically travel to the server's map and join the gameplay.
+
+With this node, you can pass in the properties `game session id` and `player id` through `AWSGameLiftJoinSessionRequest`.
+
+  
+#### AWSGameLiftJoinSessionRequest Get and Set variable nodes
+
+![AWSGameLiftJoinSessionRequest Get and Set variable nodes](/images/user-guide/gems/reference/aws/aws-gamelift/joinsessionrequest.PNG)  
+
+
+#### JoinSession node
+
+![JoinSession node](/images/user-guide/gems/reference/aws/aws-gamelift/joinsession.PNG)  
+
+
+#### JoinSessionAsync node and Async Handler node
+
+![JoinSessionAsync node and Async Handler node](/images/user-guide/gems/reference/aws/aws-gamelift/joinsessionasync.PNG)  
```

Comment:
```suggestion
#### JoinSession node

![JoinSession node](/images/user-guide/gems/reference/aws/aws-gamelift/joinsession.PNG)


#### JoinSessionAsync node and Async Handler node

![JoinSessionAsync node and Async Handler node](/images/user-guide/gems/reference/aws/aws-gamelift/joinsessionasync.PNG)


#### AWSGameLiftJoinSessionRequest Get and Set variable nodes

![AWSGameLiftJoinSessionRequest Get and Set variable nodes](/images/user-guide/gems/reference/aws/aws-gamelift/joinsessionrequest.PNG)
```
Moved the JoinSession node and JoinSessionAsync node images to the top to align with the flow/order in the section intro.